types/index.d.ts

```diff
@@ -329,8 +360,10 @@ declare module nlp {
     /**  */
     remove(): Document
     /**  */
-    hasOxfordComma(): Document
+    hasOxfordComma(): Document // Can we change all the unfinished types to 'any' instead? Not sure which ones are placeholders
```

Comment:
Fyi ^

Otherwise think this is good to go. Couldn't test plugins, so haven't updated their typings yet; how do you have the peer dep setup for testing them? npm link?

Reply:
yeah, that's fine!
I don't have any good setup

Reply:
Any idea which functions definitely return Document? I can swap the rest over to `any` then.

Build was failing for me btw `🇵🇷.` was coming out as `🇵🇷..`

text has post added to it in emoji tagger. And then again later Phrase text output (showPost). You building fine or should I make an issue?
```
  //test for unicode emojis
  if (term.text.match(emojiReg)) {
    term.tag('Emoji', 'unicode-emoji', world)
    term.text = raw
  }
```

Reply:
yeah, that's the bug I have that's blocking 12.3.
I'll get it, Just need a little coffee first.

ah, this is cool. just playing around with it now.
I may rename some variables so this screen is a little less verbose, is that okay?
![image](https://user-images.githubusercontent.com/399657/71916188-62e98200-314b-11ea-921a-a214b2903164.png)

Yeah, almost everything returns `Document`, except for `.json()` methods and ngram methods. There will be a few exceptions.

Where should these files go? in respective `./plugins` folders?

Holy, this typescript stuff is HARD. Thank you for doing this.

Reply:
Haha it's very satisfying when it works! Can probably hide some of that inside a type. I'll take a look tomorrow.

Ah fair enough, feels like the post should be cleared if it gets added to the text - or at least a flag to know it was done.

Each individual plugin just needs to export a default nlp.Plugin, I'll do the numbers plugin tomorrow to show you